daml-script/test/src/test-utils/com/daml/lf/engine/script/test/AbstractParticipantsFixture.scala

```diff
@@ -4,52 +4,67 @@
 package com.daml.lf.engine.script.test
 
 import java.io.File
-
 import com.daml.ledger.api.testing.utils.AkkaBeforeAndAfterAll
-import com.daml.lf.archive.{Dar, DarDecoder}
-import com.daml.lf.data.ImmArray
-import com.daml.lf.data.Ref.{Identifier, Name, PackageId, QualifiedName}
+import com.daml.lf.PureCompiledPackages
+import com.daml.lf.archive.DarDecoder
+import com.daml.lf.data.{Ref, ImmArray}
 import com.daml.lf.engine.script.ledgerinteraction.{ScriptLedgerClient, ScriptTimeMode}
-import com.daml.lf.engine.script.{Participants, Runner}
-import com.daml.lf.typesig.EnvironmentSignature
-import com.daml.lf.typesig.reader.SignatureReader
-import com.daml.lf.language.Ast.Package
+import com.daml.lf.engine.script.{Runner, Participants}
+import com.daml.lf.language.Ast
 import com.daml.lf.language.StablePackage.DA
-import com.daml.lf.speedy.{ArrayList, SValue}
+import com.daml.lf.speedy.{SValue, ArrayList}
 import com.daml.lf.speedy.SValue.SRecord
+import com.daml.lf.value.Value
 import org.scalatest.Suite
-import scalaz.\/-
-import scalaz.syntax.traverse._
-import spray.json.JsValue
 
 import scala.concurrent.{ExecutionContext, Future}
 
+case class CompiledDar(
+    mainPkg: Ref.PackageId,
+    compiledPackages: PureCompiledPackages,
+)
+
+object AbstractScriptTest {
+  def readDar(file: File): CompiledDar = {
+    val dar = DarDecoder.assertReadArchiveFromFile(file)
+    val pkgs = PureCompiledPackages.assertBuild(dar.all.toMap, Runner.compilerConfig)
+    CompiledDar(dar.main._1, pkgs)
+  }
+
+  def tuple(a: SValue, b: SValue) =
+    SValue.SRecord(
+      id = DA.Types.Tuple2,
+      fields = ImmArray(Ref.Name.assertFromString("_1"), Ref.Name.assertFromString("_2")),
+      values = ArrayList(a, b),
+    )
```

Comment:
I wonder if theres anywhere else we do this - seems generic enough that it could be outside of a specific test.
I thinking perhaps specifically some of the question answers for ScriptRunner, do any of those return a tuple that we manually create?